Updated documentation to address concerns in Github issue #137

aka-msft opened the pull request:
Added a few supplemental explanations to README files for cluster deployment, to provide extra detail for new users during first-time setup.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/bedrock?pullRequest=177) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/bedrock?pullRequest=177) before we can accept your contribution.<br/><br/>:x: aka-msft [sign now](https://cla.opensource.microsoft.com/Microsoft/bedrock?pullRequest=177)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/bedrock?pullRequest=177) it.</sub>

timfpark commented on cluster/README.md:
```diff
@@ -10,6 +10,12 @@ Bedrock uses three tools to automate cluster deployments that you'll need to ins
 
+Verify that these tools are added to your system's PATH in order to avoid errors during cluster deployment.  (Note to WSL users: You will need to either move the executables to /usr/local/bin, or modify ~/.bashrc to add them to PATH and persist the new additions across restarts.)
+
+In addition, you need the Azure `az` command line tool in order to create and fetch Azure configuration info:
```
Let's keep this portion in the cluster/azure subsection since this is strictly only an Azure requirement (if we do a multi-cloud deployment with GCP they do not technically need this for the GCP part)

aka-msft replied:
Moved it back.  Great point - I was grouping install tools together and didn't realize that was pulling az cli out of the "Azure" section at first.

timfpark commented on cluster/azure/README.md:
````diff
@@ -73,6 +69,8 @@ $ az ad sp create-for-rbac --subscription <id | name>
 ```
 
+Note: You may receive an error if you do not have sufficient permissions on your Azure subscription to create a service principal.
+
````
Great point - we run into this on every hackfest with real customer accounts.

timfpark commented:
Looks great @aka-msft!  My only comment is that let's move the Azure related tooling back into the `azure` folder.  Perhaps make a general comment instead that the user should read through the prereqs for the clouds they are using for their deployment.

aka-msft commented:
Thanks Tim!  Yes, I didn’t think about that when I made the change – I was thinking it would be easier to have all the necessary tools in one spot.  But you’re right that it makes more sense to move it back into the Azure section, based on the way the docs have been re-structured to allow for future additions of other cloud providers.  I’ll make that change this morning and get it pushed up!

I might end up adding a little more to the doc once I get input from Mel as well, but wanted to get something pushed up for now that included my feedback from setting up bedrock a few times and that addresses the issues in #137 at least.  Definitely let me know if you have any additional comments!

From: Tim Park <notifications@github.com>
Sent: Tuesday, March 5, 2019 7:44 PM
To: Microsoft/bedrock <bedrock@noreply.github.com>
Cc: Amanda Kaufman <Amanda.Kaufman@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/bedrock] Updated documentation to address concerns in Github issue #137 (#177)


Looks great @aka-msft<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Faka-msft&data=02%7C01%7CAmanda.Kaufman%40microsoft.com%7Cd2a7845730f6400e9de508d6a1dd850a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874370122692455&sdata=GWg5jrwjnXNl63IiFHylsNmUOyltG127WRLmlOTpp9I%3D&reserved=0>! My only comment is that let's move the Azure related tooling back into the azure folder. Perhaps make a general comment instead that the user should read through the prereqs for the clouds they are using for their deployment.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2Fbedrock%2Fpull%2F177%23issuecomment-469945281&data=02%7C01%7CAmanda.Kaufman%40microsoft.com%7Cd2a7845730f6400e9de508d6a1dd850a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874370122702463&sdata=%2BGX9RDvZSm5s9GZBUlP4QPMFs%2FAap0RJJRjv4yOdSDM%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAttXQtPzU0kUD2av3pFYAgIbB-1EA4jvks5vTytSgaJpZM4bfon4&data=02%7C01%7CAmanda.Kaufman%40microsoft.com%7Cd2a7845730f6400e9de508d6a1dd850a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636874370122702463&sdata=9TQk4Q32%2BBzixNu%2B54e096SBH5cIoRnRKI%2Fhfoi46XU%3D&reserved=0>.
